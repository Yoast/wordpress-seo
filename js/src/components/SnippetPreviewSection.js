// External dependencies.
import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { StyledSection, StyledHeading, StyledSectionBase } from "yoast-components";
import { getRtlStyle } from "yoast-components";

const Section = styled( StyledSection )`
	max-width: 640px;
	
	&${ StyledSectionBase } {
		padding-left: 0;
		padding-right: 0;

		& ${ StyledHeading } {
			font-size: 14.4px;
			${ getRtlStyle( "padding-left", "padding-right" ) }: 20px;
			margin-left: ${ getRtlStyle( "0", "20px" ) };
		}
	}
`;

/**
 * Creates the Snippet Preview Section.
 *
 * @param {Object}         props               The component props.
 * @param {ReactComponent} props.children      The component's children.
 * @param {string}         props.title         The heading title.
 * @param {string}         props.icon          The heading icon.
 * @param {bool}           props.hasPaperStyle Whether the section should have a paper style.
 *
 * @returns {ReactElement} Snippet Preview Section.
 */
const SnippetPreviewSection = ( { children, title, icon, hasPaperStyle } ) => {
	return (
		<Section
			headingLevel={ 3 }
			headingText={ title }
			headingIcon={ icon }
			headingIconColor="#555"
			hasPaperStyle={ hasPaperStyle }
		>
			{ children }
		</Section>
	);
};

SnippetPreviewSection.propTypes = {
	children: PropTypes.element,
	title: PropTypes.string,
	icon: PropTypes.string,
	hasPaperStyle: PropTypes.bool,
};

SnippetPreviewSection.defaultProps = {
	hasPaperStyle: true,
};

export default SnippetPreviewSection;
