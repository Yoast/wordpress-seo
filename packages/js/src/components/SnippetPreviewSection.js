// External dependencies.
import PropTypes from "prop-types";
import styled from "styled-components";
import { StyledSection, StyledHeading, StyledSectionBase } from "@yoast/components";
import { getDirectionalStyle as getRtlStyle } from "@yoast/helpers";

const Section = styled( StyledSection )`
	&${ StyledSectionBase } {
		padding: 0;

		& ${ StyledHeading } {
			${ getRtlStyle( "padding-left", "padding-right" ) }: 20px;
			margin-left: ${ getRtlStyle( "0", "20px" ) };
		}
	}
`;

/**
 * Creates the Snippet Preview Section.
 *
 * @param {Object}         props               The component props.
 * @param {wp.Element}     props.children      The component's children.
 * @param {string}         props.title         The heading title.
 * @param {string}         props.icon          The heading icon.
 * @param {bool}           props.hasPaperStyle Whether the section should have a paper style.
 *
 * @returns {wp.Element} Snippet Preview Section.
 */
const SnippetPreviewSection = ( { children, title, icon, hasPaperStyle, shoppingData } ) => {
	return (
		<Section
			headingLevel={ 3 }
			headingText={ title }
			headingIcon={ icon }
			headingIconColor="#555"
			hasPaperStyle={ hasPaperStyle }
			shoppingData={ shoppingData }
		>
			{ children }
		</Section>
	);
};

SnippetPreviewSection.propTypes = {
	children: PropTypes.element,
	title: PropTypes.string,
	icon: PropTypes.string,
	hasPaperStyle: PropTypes.bool,
	shoppingData: PropTypes.object,
};

SnippetPreviewSection.defaultProps = {
	hasPaperStyle: true,
	shoppingData: null,
};

export default SnippetPreviewSection;
