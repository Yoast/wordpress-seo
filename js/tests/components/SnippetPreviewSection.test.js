import React from "react";
import { mount } from "enzyme";
import SnippetPreviewSection from "../../src/components/SnippetPreviewSection";

jest.mock( "../../src/containers/SnippetEditor", () => {
	return () => {
		return <div>HI!</div>;
	};
} );

jest.mock( "yoast-components", () => {
	const yoastComponents = require.requireActual( "yoast-components" );
	return {
		...yoastComponents,
		StyledSection: () => <span>yoast-components StyledSection</span>,
	};
} );

describe( "SnippetPreviewSection", () => {
	it( "renders the snippet editor inside of it", () => {
		const tree = mount(
			<SnippetPreviewSection
				title="Snippet editor"
				icon="eye"
				theme={ { isRtl: false } }
			/>
		);
		expect( tree ).toMatchSnapshot();
	} );
} );

describe( "SnippetPreviewSection RTL", () => {
	it( "renders the snippet editor inside of it", () => {
		const tree = mount(
			<SnippetPreviewSection
				title="Snippet editor"
				icon="eye"
				theme={ { isRtl: true } }
			/>
		);
		expect( tree ).toMatchSnapshot();
	} );
} );
