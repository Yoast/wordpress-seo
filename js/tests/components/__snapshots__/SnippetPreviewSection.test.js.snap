exports[`SnippetPreviewSection RTL renders the snippet editor inside of it 1`] = `
<SnippetPreviewSection
  hasPaperStyle={true}
  icon="eye"
  theme={
    Object {
      "isRtl": true,
    }
  }
  title="Snippet editor">
  <SnippetPreviewSection__Section
    hasPaperStyle={true}
    headingIcon="eye"
    headingIconColor="#555"
    headingLevel={3}
    headingText="Snippet editor">
    <StyledSection
      className="SnippetPreviewSection__Section-kWfIbs hPjctU"
      hasPaperStyle={true}
      headingIcon="eye"
      headingIconColor="#555"
      headingLevel={3}
      headingText="Snippet editor">
      <span>
        yoast-components StyledSection
      </span>
    </StyledSection>
  </SnippetPreviewSection__Section>
</SnippetPreviewSection>
`;

exports[`SnippetPreviewSection renders the snippet editor inside of it 1`] = `
<SnippetPreviewSection
  hasPaperStyle={true}
  icon="eye"
  theme={
    Object {
      "isRtl": false,
    }
  }
  title="Snippet editor">
  <SnippetPreviewSection__Section
    hasPaperStyle={true}
    headingIcon="eye"
    headingIconColor="#555"
    headingLevel={3}
    headingText="Snippet editor">
    <StyledSection
      className="SnippetPreviewSection__Section-kWfIbs hPjctU"
      hasPaperStyle={true}
      headingIcon="eye"
      headingIconColor="#555"
      headingLevel={3}
      headingText="Snippet editor">
      <span>
        yoast-components StyledSection
      </span>
    </StyledSection>
  </SnippetPreviewSection__Section>
</SnippetPreviewSection>
`;
